Update session.py

LingweiMeng opened the pull request:
Will raise `RuntimeError` and the program stop running here without retrying.
After changing this, the problem is solved.

I'm using Mac M1 Pro, btw.

Konano commented:
Where is the `RuntimeError` raised? The expectation is `ConnectionError`.

LingweiMeng commented:
> Where is the `RuntimeError` raised? The expectation error is `ConnectionError`.

Hi @Konano,

In line41, `arknights_mower/utils/device/adb_client/session.py`
```
self.sock.send(data).check_okay()
```
The `check_okay()` function will raise `RuntimeError`, output info in terminal:
```
2022-04-22 21:07:55,712 - INFO - arknights_mower/__main__.py:45 - main - Loading the configuration file: /Users/mlw/Documents/arknights_mower/config.yaml
2022-04-22 21:07:55,789 - INFO - arknights_mower/utils/log.py:76 - set_debug_mode - Start debug mode, log is stored in /Users/mlw/Documents/arknights_mower

Traceback (most recent call last):
  File "/opt/homebrew/anaconda3/bin/arknights-mower", line 8, in <module>
    sys.exit(main())
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/__main__.py", line 55, in main
    device = Device()
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/device.py", line 72, in __init__
    self.start()
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/device.py", line 76, in start
    self.control = Device.Control(self, self.client)
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/device.py", line 33, in __init__
    self.scrcpy = Scrcpy(client)
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/scrcpy/core.py", line 72, in __init__
    self.start()
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/scrcpy/core.py", line 131, in start
    self.__init_server_connection()
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/scrcpy/core.py", line 102, in __init_server_connection
    self.__video_socket = self.client.stream('localabstract:scrcpy')
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/adb_client/core.py", line 166, in stream
    return self.session().request(cmd, True).sock
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/adb_client/session.py", line 41, in request
    self.sock.send(data).check_okay()
  File "/opt/homebrew/anaconda3/lib/python3.9/site-packages/arknights_mower/utils/device/adb_client/socket.py", line 79, in check_okay
    raise RuntimeError(self.recv_response())
RuntimeError: b'closed'
```

Konano commented:
A few days ago I modified the code there, in the `dev` branch.

https://github.com/Konano/arknights-mower/commit/82a19eecf4f3337a56772258ad93a60c63a8dfc7#diff-634bbf408cadfcc733e9b1b723f8367a02b00693ee0172c90640f239b3ca4e02

Now `ConnectionError` will be caught in the wraps `stable`.

https://github.com/Konano/arknights-mower/commit/82a19eecf4f3337a56772258ad93a60c63a8dfc7#diff-8a74ba4c8f691fa2f5704a141f6bf45520e70b33af734b7c06ec8dc8f25d59a3

I haven't pushed the code on the `dev` branch to the `main` branch yet, but I think maybe the change there solved this issues successfully?

Konano commented:
And to prevent errors when connecting before the server is fully up, I have also added some waiting interval.

https://github.com/Konano/arknights-mower/blob/bf53fc7c4ea1e85b77b7524e18fa54d4fb5b0da4/arknights_mower/utils/device/scrcpy/core.py#L140-L144

LingweiMeng commented:
> And to prevent errors when connecting before the server is fully up, I have also added some waiting interval.
> 
> https://github.com/Konano/arknights-mower/blob/bf53fc7c4ea1e85b77b7524e18fa54d4fb5b0da4/arknights_mower/utils/device/scrcpy/core.py#L140-L144

I added this time-waiting in this place, and it works now.
Thanks for sharing!
We may close this pull request now.   :)
